src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.4.0.0.cs

```diff
@@ -24,30 +19,34 @@ public partial class ExpectedExceptionTests : ConditionalWcfTest
     public static void NonExistentAction_Throws_ActionNotSupportedException()
     {
         string exceptionMsg = "The message with Action 'http://tempuri.org/IWcfService/NotExistOnServer' cannot be processed at the receiver, due to a ContractFilter mismatch at the EndpointDispatcher. This may be because of either a contract mismatch (mismatched Actions between sender and receiver) or a binding/security mismatch between the sender and the receiver.  Check that sender and receiver have the same contract and the same binding (including security requirements, e.g. Message, Transport, None).";
-        try
+        BasicHttpBinding binding = null;
+        EndpointAddress endpointAddress = null;
+        ChannelFactory<IWcfService> factory = null;
+        IWcfService serviceProxy = null;
+
+        // *** VALIDATE *** \\
+        ActionNotSupportedException exception = Assert.Throws<ActionNotSupportedException>(() =>
         {
-            BasicHttpBinding binding = new BasicHttpBinding();
-            using (ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_Basic)))
+            // *** SETUP *** \\
+            binding = new BasicHttpBinding();
+            endpointAddress = new EndpointAddress(Endpoints.HttpBaseAddress_Basic);
+            factory = new ChannelFactory<IWcfService>(binding, endpointAddress);
+            serviceProxy = factory.CreateChannel();
+
+            // *** EXECUTE *** \\
+            try
             {
-                IWcfService serviceProxy = factory.CreateChannel();
                 serviceProxy.NotExistOnServer();
             }
-        }
-        catch (Exception e)
-        {
-            if (e.GetType() != typeof(System.ServiceModel.ActionNotSupportedException))
+            finally
             {
-                Assert.True(false, string.Format("Expected exception: {0}, actual: {1}", "ActionNotSupportedException", e.GetType()));
+                // *** ENSURE CLEANUP *** \\
+                ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
             }
+        });
 
-            if (e.Message != exceptionMsg)
-            {
-                Assert.True(false, string.Format("Expected Fault Message: {0}, actual: {1}", exceptionMsg, e.Message));
-            }
-            return;
-        }
-
-        Assert.True(false, "Expected ActionNotSupportedException exception, but no exception thrown.");
+        // *** ADDITIONAL VALIDATION *** \\
+        Assert.Equal(exception.Message, exceptionMsg);
```

Comment:
Nit: I'd probably make this an Assert.True(String.Equals(), String.Format("Expected exception message {0} but actual was {1}... just to help diagnose in a lab failure.